bin/lisky

```diff
@@ -14,69 +14,112 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-var os = require('os');
-var lockfile = require('lockfile');
-var semver = require('semver');
-
-var packageJSON = require('../package.json');
-
-var configLockfilePath;
-var lisky;
-var execFile;
-var firstCommandWords;
-var nonInteractiveLiskyArg = process.argv[1];
-var nonInteractiveCommandArg = process.argv[2];
-var nonInteractiveOptions;
-var commandArgIsFilePath = false;
-
-process.env.LISKY_CONFIG_DIR = process.env.LISKY_CONFIG_DIR || `${os.homedir()}/.lisky`;
-configLockfilePath = `${process.env.LISKY_CONFIG_DIR}/config.lock`;
-
-process.env.NON_INTERACTIVE_MODE = !(nonInteractiveLiskyArg.endsWith('lisky') && process.argv.length === 2);
-
-function exit(code) {
-	process.exit(code || 0);
-}
-
-if (!semver.satisfies(process.version, packageJSON.engines.node)) {
-	console.error('\x1b[31m', `ERROR: Requires Node.js version ${semver.clean(packageJSON.engines.node)}, but was started with version ${semver.clean(process.version)}.`, '\x1b[0m');
-	exit();
-}
-
-// eslint-disable-next-line default-case
-switch (process.argv[2]) {
-case 'clean':
-	console.warn('\x1b[33m', 'WARNING: Attempting to remove configuration lockfile. I hope you know what you’re doing.', '\x1b[0m');
-	lockfile.unlockSync(configLockfilePath);
-	exit();
-	break;
-case '--version':
-case '-v':
-	console.info(packageJSON.version);
-	exit();
-	break;
-default:
-	// continue...
-}
-
-lisky = require('../dist').default;
-execFile = require('../dist/execFile').default;
-
-// eslint-disable-next-line no-underscore-dangle
-firstCommandWords = lisky.commands.map(c => c._name.split(' ')[0]);
-
-if (firstCommandWords.indexOf(nonInteractiveCommandArg) === -1) {
-	commandArgIsFilePath = true;
+const os = require('os');
+const lockfile = require('lockfile');
+const semver = require('semver');
+const packageJSON = require('../package.json');
+
+// eslint-disable-next-line import/extensions,import/no-unresolved
+const lisky = require('../dist').default;
+// eslint-disable-next-line import/extensions,import/no-unresolved
+const execFile = require('../dist/execFile').default;
+
+const nonInteractiveLiskyArg = process.argv[1];
+const nonInteractiveCommandArg = process.argv[2];
+const configLockfilePath = `${process.env.LISKY_CONFIG_DIR}/config.lock`;
+
+const errorNodeVersion = (expected, actual) =>
+	Error(`ERROR: Requires Node.js version ${expected}, but was started with version ${actual}.`);
+
+const exit = code => process.exit(code || 0);
+
+const showWarn = msg => console.warn('\x1b[33m', msg, '\x1b[0m');
+const showError = error => console.error('\x1b[31m', error.message, '\x1b[0m');
```

Comment:
Maybe it would make more sense to pass `error.message` to be more consistent with `showWarn`?